webui/src/pages/repositories/services/import_data.jsx

```diff
@@ -0,0 +1,209 @@
+import {branches, commits, config, metaRanges, NotFoundError, ranges} from "../../../lib/api";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import {LinearProgress} from "@mui/material";
+import React from "react";
+import {Link} from "react-router-dom";
+import Button from "react-bootstrap/Button";
+import Alert from "react-bootstrap/Alert";
+import Form from "react-bootstrap/Form";
+
+const ImportPhase = {
+    NotStarted: 0,
+    InProgress: 1,
+    Completed: 2,
+}
+
+const runImport = async (updateImportState, prependPath, commitMsg, sourceRef, branch, repoId, refId) => {
+    let paginationResp = {};
+    let after = "";
+    let importBranchResp;
+    let sum = 0;
+    const rangeArr = [];
+    const importStatusUpdate = {
+        importPhase: ImportPhase.InProgress,
+        numObj: sum,
+    }
+    updateImportState(importStatusUpdate);
+    do {
+        const response = await ranges.createRange(repoId, sourceRef, after, prependPath, paginationResp.continuation_token);
+        rangeArr.push(response.range);
+        paginationResp = response.pagination;
+        after = paginationResp.last_key;
+        sum += response.range.count;
+        importStatusUpdate.numObj = sum
+        updateImportState(importStatusUpdate);
+    } while (paginationResp.has_more);
```

Comment:
Did you test the import with pagination?

Reply:
yes